server/container/guice/cassandra-guice/src/main/java/org/apache/james/server/CassandraCleanupProbe.java

```diff
@@ -17,33 +17,24 @@
  * under the License.                                           *
  ****************************************************************/
 
-package org.apache.james.modules;
+package org.apache.james.server;
 
-import org.apache.james.backends.es.EmbeddedElasticSearch;
-import org.apache.james.mailbox.extractor.TextExtractor;
-import org.apache.james.mailbox.store.extractor.DefaultTextExtractor;
-import org.apache.james.util.Host;
+import javax.annotation.PreDestroy;
+import javax.inject.Inject;
 
-import com.google.inject.AbstractModule;
+import org.apache.james.backends.cassandra.init.CassandraTableManager;
+import org.apache.james.utils.GuiceProbe;
 
-public class CassandraJmapServerModule extends AbstractModule {
+public class CassandraCleanupProbe implements GuiceProbe {
```

Comment:
why it needs to be a Probe ?

Reply:
To force its instanciation via guice, so that it gets run upon destroy.

Reply:
Well, we could desine a new kind of registries, and affect **GuiceJamesServer** method with it specially for this purpose. But impacting important classes like this with only test concerns seems bad to me.

Or I could register a configuration perform that actually triggers the inject, but it seems even morebloated with me, and might break what I describe in https://github.com/linagora/james-project/pull/1550#discussion_r203646979

Reply:
> But impacting important classes like this with only test concerns seems bad to me.

It's always like that one aim of writing tests is to write production code that is more flexible, better designed, etc. In this case, we lack things in James to manage lifecycle, I don't see any good reason to not fix that issue, it will just make James a better software. Our users would benefit from that because they may use James in context we don't think of today.

TL;DR
I really think we should not trick the code but do design in these kind of cases.

Reply:
Ok then let's do a cleanup registry.